src/mlpack/methods/cf/cf_impl.hpp

```diff
@@ -31,11 +31,37 @@ void ApplyFactorizer(FactorizerType& factorizer,
                      arma::mat& w,
                      arma::mat& h,
                      const typename std::enable_if_t<FactorizerTraits<
-                         FactorizerType>::UsesCoordinateList>* = 0)
+                         FactorizerType>::UsesCoordinateList
+                         && !FactorizerTraits<
+                         FactorizerType>::UsesQuicSVD>* = 0)
 {
   factorizer.Apply(data, rank, w, h);
 }
 
+// Apply QuicSVD factorizer.
+template<typename FactorizerType>
+void ApplyFactorizer(FactorizerType& factorizer,
+                     const arma::mat& data,
+                     const arma::sp_mat& /* cleanedData */,
+                     const size_t rank,
+                     arma::mat& w,
+                     arma::mat& h,
+                     const typename std::enable_if_t<FactorizerTraits<
+                         FactorizerType>::UsesCoordinateList
+                         && FactorizerTraits<
+                         FactorizerType>::UsesQuicSVD>* = 0)
+{
+  // Get svd factorization.
+  arma::vec sigma;
+  factorizer.Apply(data, w, sigma, h);
+
+  // Sigma matrix is multiplied to w.
+  w = w * sigma;
+
+  // Take transpose of the matrix h as required by CF module.
+  h = arma::trans(h);
+}
```

Comment:
The main problem I see here is that the rank of the decomposition isn't taken into account.  One way to do this might be the following:

1. Refactor `QUIC_SVD` and `CosineTree` a little so that it terminates when enough vectors are added (see `cosine_tree.cpp:97`), and then make sure `rank` can be passed to `QUIC_SVD`.

2. Add a simple "shim" class that has the function `Apply(data, rank, w, h)` and basically calls the result of this function.

3. Ensure that the CF code uses the shim class instead of QUIC_SVD directly.

That is just one idea; maybe there are some other ideas out there, let me know if you want to do it some other way or see some better solution.